embedded-graphics/src/primitives/rounded_rectangle.rs

```diff
@@ -0,0 +1,361 @@
+//! The rectangle primitive. Also good for drawing squares.
+
+use crate::{
+    drawable::{Drawable, Pixel},
+    geometry::{Dimensions, Point, Size},
+    pixelcolor::PixelColor,
+    primitives::{
+        ellipse_quadrant::{self, EllipseQuadrant, Quadrant, StyledEllipseQuadrantIterator},
+        rectangle::{self, Rectangle, StyledRectangleIterator},
+        ContainsPoint, Primitive,
+    },
+    style::{PrimitiveStyle, Styled},
+    transform::Transform,
+    DrawTarget,
+};
+
+/// Rounded rectangle primitive
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct RoundedRectangle {
+    /// The base rectangle
+    pub rectangle: Rectangle,
+
+    corners: [EllipseQuadrant; 4],
+}
+
+impl Primitive for RoundedRectangle {
+    type PointsIter = Points;
+
+    fn points(&self) -> Self::PointsIter {
+        Points::new(self)
+    }
+}
+
+impl ContainsPoint for RoundedRectangle {
+    fn contains(&self, point: Point) -> bool {
+        unimplemented!()
+    }
+}
+
+impl Dimensions for RoundedRectangle {
+    fn bounding_box(&self) -> Rectangle {
+        self.rectangle
+    }
+}
+
+impl RoundedRectangle {
+    /// Creates a new rounded rectangle from a base rectangle and equal corner XY radius for  all
+    /// corners.
+    pub fn new(rectangle: Rectangle, corner_radius: Size) -> Self {
+        Self::with_corners(
+            rectangle,
+            [corner_radius, corner_radius, corner_radius, corner_radius],
+        )
+    }
+
+    /// Creates a new rounded rectangle with different corner radii.
+    ///
+    /// Corner radii are specified from the top-left corner in a clockwise direction
+    pub fn with_corners(rectangle: Rectangle, corner_radii: [Size; 4]) -> Self {
```

Comment:
This is a confusing name, because we also have `Rectangle::with_corners` which does something different.

Reply:
Good point. I've changed it to `with_radii`. Any ideas for a better name?

Reply:
Maybe rename `with_corners` to `new` and rename the current `new` to `with_equal` or `with_equal_radii`?

Reply:
I've gone with `with_equal_corners` as `with_equal_radii` might be misunderstood as a rounded rectangle with circular corners (i.e. `equal` meaning equal X and Y radius).

Reply:
SGTM :+1: